coderd/database/dbfake/dbfake.go

```diff
@@ -37,6 +37,64 @@ func Workspace(t testing.TB, db database.Store, seed database.Workspace) databas
 	return dbgen.Workspace(t, db, seed)
 }
 
+func TemplateWithVersion(t testing.TB, db database.Store, tpl database.Template, tv database.TemplateVersion, job database.ProvisionerJob, resources ...*sdkproto.Resource) (database.Template, database.TemplateVersion) {
+	t.Helper()
+
+	template := dbgen.Template(t, db, tpl)
+
+	tv.TemplateID = dbgen.TakeFirst(tv.TemplateID, uuid.NullUUID{UUID: template.ID, Valid: true})
+	tv.OrganizationID = dbgen.TakeFirst(tv.OrganizationID, template.OrganizationID)
+	tv.CreatedBy = dbgen.TakeFirst(tv.CreatedBy, template.CreatedBy)
+	version := TemplateVersion(t, db, tv, job, resources...)
+
+	err := db.UpdateTemplateActiveVersionByID(dbgen.Ctx, database.UpdateTemplateActiveVersionByIDParams{
+		ID:              template.ID,
+		ActiveVersionID: version.ID,
+		UpdatedAt:       dbtime.Now(),
+	})
+	require.NoError(t, err)
+
+	return template, version
+}
+
+func TemplateVersion(t testing.TB, db database.Store, tv database.TemplateVersion, job database.ProvisionerJob, resources ...*sdkproto.Resource) database.TemplateVersion {
+	templateVersion := dbgen.TemplateVersion(t, db, tv)
+	payload, err := json.Marshal(provisionerdserver.TemplateVersionImportJob{
+		TemplateVersionID: templateVersion.ID,
+	})
+	require.NoError(t, err)
+
+	job.ID = dbgen.TakeFirst(job.ID, templateVersion.JobID)
+	job.OrganizationID = dbgen.TakeFirst(job.OrganizationID, templateVersion.OrganizationID)
+	job.Input = dbgen.TakeFirstSlice(job.Input, payload)
+	job.Type = dbgen.TakeFirst(job.Type, database.ProvisionerJobTypeTemplateVersionImport)
+	job.CompletedAt = dbgen.TakeFirst(job.CompletedAt, sql.NullTime{
+		Time:  dbtime.Now(),
+		Valid: true,
+	})
+
+	job = dbgen.ProvisionerJob(t, db, nil, job)
+	ProvisionerJobResources(t, db, job.ID, "", resources...)
+	return templateVersion
+}
+
+func TemplateVersionWithParams(t testing.TB, db database.Store, tv database.TemplateVersion, job database.ProvisionerJob, params []database.TemplateVersionParameter) (database.TemplateVersion, []database.TemplateVersionParameter) {
```

Comment:
You've got `TemplateVersion` (which accepts optional resources) and `TemplateVersionWithParams` (which accepts params, but no resources).  And, most of the time you pass an empty `database.TemplateVersion` and `database.ProvisionerJob`.  What do you think of making the version, job, resources and parameters optional parameters, either with `...opts` or using a Builder pattern?